src/sources/util/grpc/mod.rs

```diff
@@ -43,6 +49,9 @@ where
 
     info!(%address, "Building gRPC server.");
 
+    // Conditionally apply the ConnectionLimitLayer if any limits are set
+    let service = ConnectionLimitLayer::new(max_requests, max_duration).layer(service);
```

Comment:
I think this could be added in the list before as another layer after (or before) the DecompressionAndMetricsLayer.